app/src/main/java/com/github/se/travelpouch/model/activity/ActivityViewModel.kt

```diff
@@ -0,0 +1,128 @@
+package com.github.se.travelpouch.model.activity
+
+import android.content.Context
+import android.util.Log
+import android.widget.Toast
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.github.se.travelpouch.model.Location
+import com.google.firebase.Firebase
+import com.google.firebase.Timestamp
+import com.google.firebase.firestore.firestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class ActivityModelView(val activityRepositoryFirebase: ActivityRepository) : ViewModel() {
+
+  // create factory
+  companion object {
+    val Factory: ViewModelProvider.Factory =
+        object : ViewModelProvider.Factory {
+          @Suppress("UNCHECKED_CAST")
+          override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            return ActivityModelView(ActivityRepositoryFirebase(Firebase.firestore)) as T
+          }
+        }
+  }
+
+  private val activities_ = MutableStateFlow<List<Activity>>(emptyList())
+  val activities: StateFlow<List<Activity>> = activities_.asStateFlow()
+
+  private val selectedActivity_ =
```

Comment:
Why did you initialize selectedActivity_ with a default Activity  instead of setting it to null?